source/agora/utils/TxBuilder.d

```diff
@@ -159,7 +162,7 @@ public struct TxBuilder
         this.inputs ~= iota(tx.outputs.length)
             .map!(index => OutputRef(tx.outputs[index], Input(tx, index).utxo))
             .array;
-        if (tx.outputs.any!(outp => !this.leftover.value.add(outp.value)))
+        if (this.inputs.any!(input => !this.leftover.value.add(input.output.value)))
```

Comment:
Why this change ? The reason we were iterating on `tx.outputs` is to be able to do `builder.attach(txa).attach(txb)`.
AFAICS this would break now ?

Reply:
The `this.inputs` are made from the `tx.outputs` just above this line so I do not see it making any difference to the chaining.

Reply:
On the first call, `this.inputs` and `tx.outputs` are the same, on the second call, `this.inputs` will not be empty.

Reply:
Yes of course. I'll update.